tests/integration/ha_tests/helpers.py

```diff
@@ -164,25 +169,25 @@ async def count_primaries(ops_test: OpsTest) -> int:
     wait=wait_exponential(multiplier=1, min=2, max=30),
 )
 async def replica_set_primary(
-    replica_set_hosts: List[str], ops_test: OpsTest, return_name=False
+    replica_set_hosts: List[str],
+    ops_test: OpsTest,
+    down_unit=None,
+    app=None,
 ) -> str:
```

Comment:
You have a type-hint here for `str`, whereas on L190 you return type `Unit`, and depend on type `Unit` on L468 is that right?

Reply:
And you return `None` on L186, so you'll want to hint `Optional` too.